prow/tide/tide.go

```diff
@@ -275,6 +282,83 @@ func (c *Controller) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// filterSubpools filters non-pool PRs out of the initially identified subpools,
+// deleting any pools that become empty.
+// See filterSubpool for filtering details.
+func (c *Controller) filterSubpools(raw map[string]*subpool) map[string]*subpool {
+	filtered := make(map[string]*subpool)
+	for key, sp := range raw {
+		var err error
+		// TODO: move initialization of 'presubmits' 'cc' and 'sha' to an 'initPool' func.
+		sp.presubmits, err = c.presubmitsByPull(sp)
+		if err != nil {
+			sp.log.WithError(err).Error("Determining required presubmit prowjobs.")
+			continue
+		}
+		sp.cc, err = c.ca.Config().GetTideContextPolicy(sp.org, sp.repo, sp.branch)
+		if err != nil {
+			sp.log.WithError(err).Error("Setting up context checker.")
+			continue
+		}
+
+		if spFiltered := filterSubpool(c.ghc, sp); spFiltered != nil {
+			filtered[key] = spFiltered
+		}
+	}
+	return filtered
+}
+
+// filterSubpool filters PRs from an initially identified subpool. If the subpool
+// becomes empty 'nil' is returned.
+// Specifically we filter out PRs that:
+// - Have known merge conflicts.
+// - Have failing or missing status contexts.
+// - Have pending status contexts that are not associated with a required ProwJob.
```

Comment:
Will this break merging PRs that don't use prow for testing?

Reply:
Won't it just wait until the non-Prow controlled status is resolved to something other than `pending`?

Reply:
Exactly what Steve said. I figured we shouldn't consider those PRs part of the pool until the statuses that are out of our control (but are required) are all green since Tide won't be able to merge until they are green and Tide can't retrigger those tests itself.